### PR TITLE
Refactor parsing logic

### PR DESCRIPTION
**Pull request by fadushin**
This PR refactors the parsing logic in our providers, making use of rebar3 internal parsing code.

There should be no major functional differences in this code.  Mostly cleanup.

### CODE REVIEW

File: src/atomvm_esp32_flash_provider.erl [hunk 1]
```diff
@@ -25,11 +25,11 @@
 -define(PROVIDER, esp32_flash).
 -define(DEPS, [packbeam]).
 -define(OPTS, [
-    {esptool, $e, "esptool", undefined, "Path to esptool.py"},
-    {chip, $c, "chip", undefined, "ESP chip (default auto)"},
-    {port, $p, "port", undefined, "Device port (default /dev/ttyUSB0)"},
-    {baud, $b, "baud", undefined, "Baud rate (default 115200)"},
-    {offset, $o, "offset", undefined, "Offset (default 0x210000)"}
+    {esptool, $e, "esptool", string, "Path to esptool.py"},
```
Comment (UncleGrumpy) [inline src/atomvm_esp32_flash_provider.erl:28]: Is the fourth argument here "string" used to validate user input?

File: src/legacy_stm32_flash_provider.erl [hunk 2]
```diff
@@ -24,10 +24,10 @@
 -export([init/1, do/1, format_error/1]).
 
 -define(PROVIDER, stm32_flash).
--define(DEPS, [packbeam]).
+-define(DEPS, [{default, app_disovery}]).
```
Comment (UncleGrumpy) [inline src/legacy_stm32_flash_provider.erl:27]: I don't now if this is correct. Does `app_disovery` make sure the application has been compiled, and packed into an avm file?

File: src/atomvm_esp32_flash_provider.erl (hunk 1 above)
Comment (fadushin) [reply src/atomvm_esp32_flash_provider.erl:28]: Not so much checking as much type coercion, I think.  I might retry making baud an integer.

File: src/legacy_stm32_flash_provider.erl (hunk 2 above)
Comment (fadushin) [reply src/legacy_stm32_flash_provider.erl:27]: Rats I thought I fixed that.  Will fix.

**Review by UncleGrumpy — APPROVED** [review]
Looks good.

**Pull request merged**